tests/reader/reader_test.go

```diff
@@ -1,79 +1,151 @@
 package reader_test
 
 import (
+	"io"
+	"os"
 	"strings"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
```

Comment:
You should have kept assert, and use require only where needed.

I'm a bit afraid the error messages of your tests will be uneasy to read fix when an errors will occur.

I also went through this kind of changes years ago, then I had to maintain the code

Reply:
I've tackled the problem you mentioned about the lack of description on the error messages. 

Now, which part of the tests do you think I should keep the assertion? I tried to do some updates in the latest commit, tackling this, but I feel like you meant in the following part:

```go
t.Run("file is not readable", func(t *testing.T) {
		file := tests.CreateTempFile(t, "content")

		// Remove read permissions
		err := os.Chmod(file.Name(), 0o200) // 0200 corresponds to write-only permission.
		require.NoError(t, err, "expected no error when changing file permissions to write-only")

		err = contentReader.Readable(file.Name())
    // assert.ErrorIs(t, err, console.ErrPermissionDenied, "expected permission denied error when file is not readable")
		require.ErrorIs(t, err, console.ErrPermissionDenied, "expected permission denied error when file is not readable")

		// Restore read permissions for cleanup
		err = os.Chmod(file.Name(), 0o600) // 0600 corresponds to read-write permissions.
		require.NoError(t, err, "expected no error when restoring file permissions")
	})
```